EFCore.CheckConstraints/Internal/EnumCheckConstraintConvention.cs

```diff
@@ -54,41 +55,63 @@ public virtual void ProcessModelFinalizing(IConventionModelBuilder modelBuilder,
                     continue;
                 }
 
-                var enumValues = Enum.GetValuesAsUnderlyingType(propertyType).Cast<object>().Distinct().ToArray();
-                if (enumValues.Length == 0)
+                if (!_cachedConstraints.TryGetValue(typeMapping, out var constraintSql))
```

Comment:
Would it be possible to add a test to check that we're not generating more than necessary? I know checking internals is messy but if the point is to cache correctly, we should add a test too to ensure we're not throwing in code that's not actually doing what we want 😁

Reply:
Yeah, it's a bit much for me. I've manually confirmed (by debugging through) that the caching works - adding an automated test for that, with the required exposing of internals, seems like a bit of over-defensive engineering...

Reply:
You wouldn't have to actually expose them, you would use reflection to grab the private field and ensure there's only one entry in the dictionary. I'm not gonna push harder on it, I just want to make sure things are working as intended.

Reply:
That kind of internal access adds burden, since the test breaks as you refactor thing (e.g. rename the field or whatever). I usually restrict doing that to cases where it's really important - this doesn't really feel like that...